Developers-Notes/conp-dataset_creation_of_prezipped_archives_for_downloads.md

```diff
@@ -0,0 +1,153 @@
+
+To create the archive, run `datalad export-archive` in the `projects` directory of `conp-dataset` 
+with the following options:
+
+```
+datalad export-archive -d <dataset directory> -t tar -c '' <path to output directory>/<dataset directory name>.tar
+```
+
+Example for the mousebytes dataset and the the output directory `/data-scratch1/for_CONP_portal/.cache`:
+
+```
+datalad export-archive -d mousebytesdataset -t tar -c '' /data-scratch1/for_CONP_portal/.cache/mousebytesdataset.tar
+```
+
+Notes: 
+
+- running `datalad export-archive` this way ensures that the dataset name is in the name of the 
+archive and that the extracted dataset will be done in folder named like the dataset instead of the DataLad ID.
+- `''` is given to the option `c` so that the produced archive is not gzip since we will need to modify
+the archive in point 7. (An archive cannot be modified if it is gzipped.)
+
+##### 6. Create a file with the git log history of the dataset 
+
+Once the DataLad archive has been exported run the following command to export the git history:
+
+```
+cd <dataset name>
+mkdir <path to output directory>/<dataset name>
+git log --pretty=format:"%H %s" > <path to output directory>/<dataset name>/.<dataset name>.git.log>
+cd ..
+```
+
+Example for the mousebytes dataset and the output directory `/data-scratch1/for_CONP_portal/.cache`:
+
+```
+cd mousebytes
+mkdir /data-scratch1/for_CONP_portal/.cache/mousebytes
+git log --pretty=format:"%H %s" > /data-scratch1/for_CONP_portal/.cache/mousebytes/.mousebytes.git.log>
+cd ..
+```
+
+Note: to run this step on a list of datasets, create a text file with the list of datasets to process
+and run:
+
+```
+cat ../../list_of_datasets.txt | while read f; do cd $f; mkdir -p /data-scratch1/for_CONP_portal/.cache/$f; git log --pretty=format:"%H %s" > /data-scratch1/for_CONP_portal/.cache/$f/.$(basename $f).git.log; cd -; done
+```
+
+##### 7. Update the extracted archive to append the .git.log file
+
+Once the DataLad archive has been created, run the script `XXXXXXX` that will:
+
+```
+cd <the output directory>
+tar -rf <dataset name>.tar <dataset name>/.<dataset name>.git.log
+gzip <dataset name>.tar
+```
+
+Example for the mousebytes dataset in the output directory `/data-scratch1/for_CONP_portal/.cache`:
+
+```
+cd /data-scratch1/for_CONP_portal/.cache
+tar -rf mousebytes.tar mousebytes/.mousebytes.git.log
+gzip mousebytes.tar
+```
+
+Note: to run this step on a list of datsets, create a text file with the list of datasets to process
+and run the following in the directory that contains the archives:
+
+```
+cat ../list_of_datasets.txt | while read f; do tar -rf $(basename $f).tar $(basename $f)/.$(basename $f).git.log; gzip $(basename $f).tar; done
+```
+
+##### 8. Add the dataset version to the archive filename
+
+This was done by hand but could be scripted in future creation of prezipped
+archives.
+
+```
+mv <dataset name>.tar.gz <dataset_name>_version-<version-number>.tar.gz
+```
```

Comment:
@xlecours Not sure I understand your latest comment. The renaming with the proper version is done in step 8. here. Am I missing something?

Reply:
@xlecours I just pushed some changes that maybe will reply to your comments about he _version- part?

Reply:
```suggestion
mv <dataset name>.tar.gz <dataset_name>_version-<version-string>.tar.gz
Where <version-string> is the value of the version property in the DATS.json file.
```